Recording page

Alec-Hallman opened the pull request:
### Created the recording page. 

Users can click the microphone button to start a recording, clicking it again will pause the recording and display a little "saved" icon at the bottom of the screen. Whilst audio is being recorded, a sound wave will appear in the blue box displaying their mic input and the top box shows a sound speed clock as well as the recording status. When the recording page or app is closed, the audio file is accessible in your phone's download folder. The name of the file correlates to the date and time at which the recording was started. This page also features the bottom navigation bar (thanks Anika) and can traverse back and forth between the home page. A little "are you sure?" pop up will appear if the user tries to go to the homepage while recording.

> Notes
The recording page does not feature a stop/save button, as the audio recorder will automatically do so. Instead, I created a little pop-up to communicate when the recording is saved. Any feedback on this would be appreciated, as this still might not be as intuitive as it could be. But I enjoy how un intrusive this system is, require the user to press less buttons.

![image](https://github.com/LaurierCS/POD1/assets/109433535/cafc1dd7-f441-4263-88bb-7f04d7561d49)

Alec-Hallman committed:
Base functionality

Added:
-App bar (soon to be removed)
-Top container displaying the recording speed time and the current status of the page
-Bottom container which displays the soundwaves being captured from the phone microphone.
-Floating action button which controls the playing and pausing of the recoding.

Alec-Hallman committed:
Merge remote-tracking branch 'origin/Anika/home' into Alec/RecordingPage

Alec-Hallman committed:
Home page access, and general cleaning

- Now goes from homepage to the record page when the + button is clicked. Uses Flutters Navigator.push system.
- Added a little saved pop up which appears when ever the recording is paused or played.
- Cleaned up the code removing unused old chunks and cleaned up imports.
- Added the bottom row to the page to make it the same as the home page.
- Files still save to storage in download, as I can't find the app directory on my phone or laptop, still. But the file name now corelates to the date and time the recording was started.

pabloj2001 commented on mobile/voice_journal_app/android/app/src/debug/AndroidManifest.xml:
```diff
@@ -4,4 +4,5 @@
          to allow setting breakpoints, to provide hot reload, etc.
     -->
     <uses-permission android:name="android.permission.INTERNET"/>
+    <uses-permission android:name="android.permission.RECORD_AUDIO" />
```
While declaring permissions here works for Android, we won't get the permission on iOS devices. A better way to do it is to ask for permissions through Flutter when the record button is clicked: https://stackoverflow.com/a/63480023/4749973

pabloj2001 commented on mobile/voice_journal_app/lib/RecordingPage.dart:
```diff
@@ -0,0 +1,250 @@
+        ),
+      ),
+    floatingActionButtonLocation: FloatingActionButtonLocation.centerDocked, // Centers the button above the bar
+    bottomNavigationBar: BottomAppBar( //Bottom bar yoinked from Anika's home page (Thanks)
```
We generally want to avoid having duplicates of code in our project and instead want to reuse as much as possible. Instead of putting the bottom nav here, we'll want to put it in the main file and show the pages as widgets within it.

pabloj2001 commented on mobile/voice_journal_app/lib/home.dart:
```diff
@@ -1,43 +1,72 @@
 import 'package:flutter/material.dart';
-
+import 'RecordingPage.dart'; //Importing recording page
```
We seem to be merging changes from other PRs here, this is generally bad practice. I know we need to update this page to add the navigation to the recording page, but let's hold off on it until Anika's home page changes get merged. Let's remove this page for now. Sorry if I made it confusing!

pabloj2001 commented on mobile/voice_journal_app/lib/Pages/Alec_Recording.dart:
```diff
@@ -0,0 +1,19 @@
+import 'package:flutter/material.dart';
```
I don't think we need another page as a starting point for the app. Let's remove this page if there's no need for it.

pabloj2001 commented on mobile/voice_journal_app/lib/RecordingPage.dart:
```diff
@@ -0,0 +1,250 @@
+      home: const RecordingPage(title: "recordingPage"),
+    );
+  }
+}
```
There's no need for this starting point in the app since we already have `main.dart` and `RecordingPage` will be called when navigated to.

pabloj2001 reviewed (changes requested):
Good job on this first PR! There's a few things I think we need to update but we're on the right track.

Alec-Hallman committed:
Tunned recording page

Applied Pablo's suggestions.
- Removed Anika's homepage
- Removed void main Main app
-Removed Alec/Recording page file (meant to delete this a few days ago but forgot)
-Removed forced Android permissions

Note: The permission call built into the recording library works for me. Unsure about IOS but I think it should still work for both. So I didn't need to add the permission library/flutter call as suggested.

Alec-Hallman committed:
re-added home.dart

Alec-Hallman committed:
Merge branch 'main' into Alec/RecordingPage

Alec-Hallman committed:
Fixing Errors:

pabloj2001 reviewed (approved):
Looks Good to Me!

anikapatychuk commented on mobile/voice_journal_app/lib/RecordingPage.dart:
not sure if I looked at this at all right but I had a problem were as soon I hit record it paused me and the wouldn't un-pause

pabloj2001 committed:
Merge branch 'main' into Alec/RecordingPage

TaseskiCS reviewed (commented):
Just tested it, also had the unpause issue, but other than that everything works great!
